src/commands/basic/setup.js

```diff
@@ -0,0 +1,91 @@
+const ConfigModel = require("../../database/schemas/config")
+
+module.exports = {
+    config: {
+        name: "setup",
+        aliases: ["config"],
+        usage: "",
+        category: "basic",
+        description: "Bot Server setup",
+        accessibility: "Moderator",
+    },
+    run: async (client, message, args) => {
+        const messageFilter = response => {
+            return response.author.id === message.author.id
+        }
+        message.channel.send("Hello! This is the bot setup process for this server")
+        message.channel.send("Please type in a contact for this server or the server's owner")
+        const contact = (await message.channel.awaitMessages(messageFilter, { max: 1, time: 30000 })).first()?.content
+        if (contact === undefined) return message.channel.send("Didn't send contact in time")
+        message.channel.send("Please ping (or type in the ID of) your role of people which can create violations")
+        let role = (await message.channel.awaitMessages(messageFilter, { max: 1, time: 30000 })).first()
+        if (role.mentions.roles.first()) role = role.mentions.roles.first().id
+        else role = role.content
+        if (message.guild.roles.cache.get(role) === undefined) return message.channel.send("Role is not correct")
+        message.channel.send("Please type in your API key that you recieved from the FAGC team, if you did recieve one. Type `none` if you didn't")
+        let apikey = (await message.channel.awaitMessages(messageFilter, { max: 1, time: 30000 })).first()
+        apikey.delete()
+        if (apikey.content === 'none') apikey = undefined
+        else apikey = apikey.content
+        
+        let embed = new MessageEmbed()
+            .setTitle("FAGC Config")
+            .setAuthor("FAGC Community")
+            .setTimestamp()
+            .setDescription("Your FAGC Configuration")
+        embed.addFields(
+            {name: "Community name", value: message.guild.name},
+            {name: "Contact", value: contact},
+            {name: "Moderator role", value: `<@&${role}>`},
+            {name: "API key", value: apikey ? "Hidden" : "None"}
+        )
+        message.channel.send(embed)
+        const confirm = await message.channel.send("Are you sure you want these settings applied?")
+        confirm.react("✅")
+        confirm.react("❌")
+        const reactionFilter = (reaction, user) => {
+            return user.id == message.author.id
+        }
+        let reactions
+        try {
+            reactions = await confirm.awaitReactions(reactionFilter, { max: 1, time: 120000, errors: ['time'] })
+        } catch (error) {
+            return message.channel.send("Timed out.")
+        }
+        let reaction = reactions.first()
+        if (reaction.emoji.name === "❌")
+            return message.channel.send("Community configuration cancelled")
+        try {
+            const originalData = await ConfigModel.findOne({
```

Comment:
I understand `message.guild.id` What do you mean with the `exists`? I assume you are talking about the MongoDB `$exists` operator, but I'm not sure how you could make use of that here.

Reply:
You would use `exists` in place of your test on line 64 `if (originalData) {` and remove the call to `findOne` as I would expect `exists` to be more optimised. The comment on `message.guild.id` was to show that you are not required to use `findOne`

Reply:
If I were to use `exists` instead, I wouldn't be able to check *which* guild it is, only that there is one
> When <boolean> is true, $exists matches the documents that contain the field, including documents where the field value is null. If <boolean> is false, the query returns only the documents that do not contain the field.
As it would check for any documents that have the guildid field and return the first one it finds, not the specific one that I need. I could however use the following query instead:
```js
await ConfigModel.findOne({
    guildid: { $exists: true, $in: [message.guild.id]},
})
```
Not sure if this was what you mean.

Reply:
> If I were to use `exists` instead, I wouldn't be able to check which guild it is, only that there is one

I see no issue with this because you do not check which guild it is.
The condition of `if (originalData) {` is the only place you access the value, and this tests if the value exists.
Checking the docs I found that `findOne` returns a query object, so I believe you missed a `.exec()`